### PR TITLE
3428 - Implement per tenant activity.log backup path

### LINKED ISSUE #3428: implement per tenant activity.log backup path (3 pts)
Right now the backups are stored on uwazi_root_path/logs/tenantName_activity.log.
we should be able to config on the tenant the path where this is stored.

the production environment has this backups all around the different core deployments paths, maybe move them to the proper location after the implementation ?

### PR DESCRIPTION
**Pull request by Zasa-san**
fixes #3428

PR checklist:
- [ ] Update READ.me ?
- [x] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

Linked to https://github.com/huridocs/uwazi-documentation/pull/11

### CODE REVIEW

File: app/api/activitylog/specs/activitylogMiddleware.spec.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
   let res;
   let next;
 
+  beforeAll(() => {
+    testingTenants.mockCurrentTenant({
+      name: 'tenant1',
+      activityLogs: `${__dirname}/../../files/specs/uploads`,
+    });
+  });
+
```
Comment (daneryl) [inline app/api/activitylog/specs/activitylogMiddleware.spec.js:19]: @Zasa-san we designed testingEnvironment to do this exactly, but the name is hardcoded there, i suggest you add support for name  use setTenant instead.
```
  beforeAll(() => {
    testingEnvironment.setTenant('tenant1');
  });

```

what do you think ?

File: app/api/config.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ const {
   UPLOADS_FOLDER,
   CUSTOM_UPLOADS_FOLDER,
   TEMPORAL_FILES_FOLDER,
+  ACTIVITY_LOGS,
```
Comment (daneryl) [inline app/api/config.ts:9]: @Zasa-san all the other variables for folders config have a ```_FOLDER``` at the end, can you change the name for the acritivy_logs var also ?

File: app/api/activitylog/activitylogMiddleware.js [hunk 3]
```diff
@@ -32,26 +34,40 @@ function mustBeLogged(baseurl, method, body) {
...
+
+export default (req, _res, next) => {
+  try {
+    const { url, method, body = {} } = req;
+    const baseurl = url.split('?').shift();
+    if (mustBeLogged(baseurl, method, body)) {
+      const entry = createEntry({ ...req, url: baseurl });
+      activitylog.save(entry);
+      appendFile(
+        activityLogPath(`${tenants.current().name}$_activity.log`),
```
Comment (daneryl) [inline app/api/activitylog/activitylogMiddleware.js:63]: @Zasa-san i think the ```$``` here is a typo ? the generated file is ```tenant$_activity.log``` ?

**Review by daneryl — CHANGES_REQUESTED** [review]
@Zasa-san looks good, functional qa working as expected.
I have written some comments and suggestions on the code.

File: app/api/activitylog/activitylogMiddleware.js (hunk 3 above)
Comment (Zasa-san) [reply app/api/activitylog/activitylogMiddleware.js:63]: Yes, its a typo. Fixed.

File: app/api/activitylog/specs/activitylogMiddleware.spec.js (hunk 1 above)
Comment (Zasa-san) [reply app/api/activitylog/specs/activitylogMiddleware.spec.js:19]: Yes, I forgot that testingEnviroment already handled file routes. Changed!

**Review by daneryl — APPROVED** [review]

**Pull request merged**